pkg/operator/controllers/checkers/ingresscertificatechecker/checker.go

```diff
@@ -59,13 +62,41 @@ func (r *checker) Check(ctx context.Context) error {
 		return err
 	}
 
-	if ingress.Spec.DefaultCertificate == nil {
+	cluster, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	return validateCertificate(cv.Spec.ClusterID, ingress.Spec.DefaultCertificate, cluster.Spec.Domain)
+}
+
+func validateCertificate(clusterId configv1.ClusterID, certificate *corev1.LocalObjectReference, clusterDomain string) error {
+	clusterHasManagedDomain := clusterHasManagedDomain(clusterDomain)
```

Comment:
Is not this redefining the logic in dns.ManagedDomain ?

Reply:
Good point.
[dns.ManagedDomain()](https://github.com/Azure/ARO-RP/blob/master/pkg/util/dns/dns.go#L193) expects as a parameter the whole env.Interface which is overkill, so after your comment, I thought it would be better to move the function which checks if a clusterDomain is managed to the dns package and make it public and directly unit-tested.

Tks!